test/monitoring/monitoring.go

```diff
@@ -52,34 +57,39 @@ func GetPods(kubeClientset *kubernetes.Clientset, app, namespace string) (*v1.Po
 	return pods, err
 }
 
-// Cleanup will clean the background process used for port forwarding
-func Cleanup(pid int) error {
-	ps := os.Process{Pid: pid}
-	return ps.Kill()
-}
-
 // PortForward sets up local port forward to the pod specified by the "app" label in the given namespace
-func PortForward(logf logging.FormatLogger, podList *v1.PodList, localPort, remotePort int, namespace string) (int, error) {
-	podName := podList.Items[0].Name
-	portFwdCmd := fmt.Sprintf("kubectl port-forward %s %d:%d -n %s", podName, localPort, remotePort, namespace)
-	portFwdProcess, err := executeCmdBackground(logf, portFwdCmd)
-
-	if err != nil {
-		return 0, fmt.Errorf("failed to port forward: %w", err)
+// To close the port forwarding, just close the channel
+func PortForward(logf logging.FormatLogger, config *rest.Config, clientSet *kubernetes.Clientset, pod *v1.Pod, localPort, remotePort int) (chan struct{}, error) {
```

Comment:
The previous version took a *v1.PodList". (I also wonder if passing the logger, clientSet, etc would be more cleanly done via injection/`context.Context`.)

Reply:
That really didn't made sense, because you port forward one pod at the time